### PR TITLE
missing puesdo code and jest will have completed by 7/19/2022

### PR DESCRIPTION
**Pull request by AriseSon**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

Missing puesdo code, jest test and interview questions.... will have completed by 7/19/2022

### CODE REVIEW

**Comment by AriseSon** [comment]
Added psuedo code and jest

**Comment by AriseSon** [comment]
"COMPLETED"

File: ruby-challenges.rb [hunk 1]
```diff
@@ -19,17 +30,55 @@
...
+my_bike = Bike.new("Trek", 2, 0)
+# "The Trek bike has 2 wheels and is going 0 mph."
+p my_bike.pedal_faster(10)
+# 10
+p my_bike.pedal_faster(18)
+# 28
+p my_bike.brake(5)
+# 23 
+p my_bike.brake(25)
+# 0
```
Comment (triciasykes) [inline ruby-challenges.rb:83]: You are Rocking Ruby, Ricky!

File: interview-questions.md [hunk 2]
```diff
@@ -10,42 +10,69 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+2. JSON: JavaScript Object Notation, more commonly known by the acronym JSON, is an open data interchange format that is both human and machine-readable. Despite the name JavaScript Object Notation, JSON is independent of any programming language and is a common API output in a wide variety of applications.
 
-3. ERB:
+3. ERB: ERB (or Ruby code generated by ERB ) returns a string in the same character encoding as the input string. When the input string has a magic comment, however, it returns a string in the encoding specified by the magic comment.
 
-4. Params:
+4. Params: params is an alias for the parameters method. params comes from ActionController::Base, which is accessed by your application via ApplicationController. Specifically, params refers to the parameters being passed to the controller via a GET or POST request.
 
-5. API:
+5. API: API is the acronym for Application Programming Interface, which is a software intermediary that allows two applications to talk to each other. Each time you use an app like Facebook, send an instant message, or check the weather on your phone, you're using an API.
```
Comment (triciasykes) [inline interview-questions.md:78]: excellent work on the interview questions.  Love your examples.  Keep up the great work!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,165 @@
...
+       return false
+     }
+
+    console.log(fullHouse(hand1))
+    // true
+    console.log(fullHouse(hand2))
+    // false
+    console.log(fullHouse(hand3))
+    // false
+
```
Comment (triciasykes) [inline code-challenges.test.js:179]: Excellent work, Ricky! I like that you showed both attempts!  Way to work through the problem!

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,42 +15,165 @@
...
+    let secretCodeWord = String.replace(/a/g, 4).replace(/A/g, 4).replace(/e/g, 3).replace(/E/g, 3).replace(/i/g, 1).replace(/I/g, 1).replace(/o/g, 0).replace(/O/g, 0)
+    return secretCodeWord
+}
+console.log(secretCodeWordCreator(secretCodeWord1))
+// L4ck4d41s1c4l
+console.log(secretCodeWordCreator(secretCodeWord2))
+// G0bbl3dyg00k
+console.log(secretCodeWordCreator(secretCodeWord3))
+// 3cc3ntr1c
+
```
Comment (triciasykes) [inline code-challenges.test.js:53]: Great work - just for future challenges...when you use regex please explain what it's doing so we know you have an understanding of it.  Well done!

**Review by triciasykes — APPROVED** [review]
Ricky!  You really crushed this assessment!  Your pseudo code and test look great and your code is clean and efficient.  Your interview questions are on point.  I can tell you put time and effort into this assessment.  You are leveling up your skills!  Excellent work!

**Pull request merged**